gamla/functional.py

```diff
@@ -366,3 +366,36 @@ def eq_by(f, value_1, value_2):
 
 
 eq_str_ignore_case = eq_by(str.lower)
+
+
+@toolz.curry
+def countby_many(f, it):
+    """Count elements of a collection by a function which returns a tuple of keys
+    for single element.
+
+    Parameters:
+    f (Callable): Key function (given object in collection outputs tuple of keys).
+    it (Iterable): Collection.
+
+    Returns:
+    Dict[Text, Any]: Dictionary where key has been computed by the `f` key function
+    and value is the frequency of this key.
+
+    >>> names = ['alice', 'bob', 'charlie', 'dan', 'edith', 'frank']
+    >>> countby_many(lambda name: (name[0], name[-1]), names)
+    {'a': 1,
+     'e': 3,
+     'b': 2,
+     'c': 1,
+     'd': 1,
+     'n': 1,
+     'h': 1,
+     'f': 1,
+     'k': 1}
+    """
+    return toolz.pipe(
+        it,
+        curried.mapcat(f),
+        curried.groupby(toolz.identity),
```

Comment:
consider reduce_by_many for memory reasons

Reply:
do you mean `groupby_many`? there is no reduce_by_many in gamla

Reply:
* `groupby_many_reduce` ?

Reply:
used `groupby_many_reduce`